modules/swagger-codegen/src/main/resources/go/api.mustache

```diff
@@ -42,15 +41,27 @@ func (a {{classname}}) {{nickname}} ({{#allParams}}{{paramName}} {{{dataType}}}{
   var httpMethod = "{{httpMethod}}"
  // create path and map variables
   path := a.Configuration.BasePath + "{{path}}"
-{{#pathParams}} path = strings.Replace(path, "{" + "{{baseName}}" + "}", fmt.Sprintf("%v", {{paramName}}), -1)
+{{#pathParams}}  path = strings.Replace(path, "{" + "{{baseName}}" + "}", fmt.Sprintf("%v", {{paramName}}), -1)
 {{/pathParams}}
 
   {{#allParams}}
   {{#required}}
   // verify the required parameter '{{paramName}}' is set
-  if &{{paramName}} == nil {
-      return {{#returnType}}*new({{{returnType}}}), {{/returnType}}*NewAPIResponseWithError("400 - Bad Request"), errors.New("Missing required parameter '{{paramName}}' when calling {{classname}}->{{operationId}}")
+  {{#isListContainer}}
+  if len({{paramName}}) == 0 {
+       return {{#returnType}}*new({{{returnType}}}), {{/returnType}}*NewAPIResponseWithError("400 - Bad Request"), fmt.Errorf("missing required parameter '{{paramName}}' when calling {{classname}}->{{operationId}}")
```

Comment:
Does Go have some sort of built-in exception for invalid argument? Similar to [this](http://stackoverflow.com/a/774116/677735) in C#